connector/src/main/java/org/geysermc/connector/network/translators/block/BlockTranslator.java

```diff
@@ -101,6 +102,50 @@
                 JAVA_ID_TO_BLOCK_ENTITY_MAP.put(javaBlockState, javaId);
             }
 
+            // If the Java ID is bed, signal that it needs a tag to show color
+            // The color is in the namespace ID in Java Edition but it's a tag in Bedrock.
+            byte bedcolor = -1;
+            if (javaId.contains("_bed[")) {
+                if (javaId.contains("minecraft:white")) {
+                    bedcolor = 0;
+                } else if (javaId.contains("minecraft:orange")) {
+                    bedcolor = 1;
+                } else if (javaId.contains("minecraft:magenta")) {
+                    bedcolor = 2;
+                } else if (javaId.contains("minecraft:light_blue")) {
+                    bedcolor = 3;
+                } else if (javaId.contains("minecraft:yellow")) {
+                    bedcolor = 4;
+                } else if (javaId.contains("minecraft:lime")) {
+                    bedcolor = 5;
+                } else if (javaId.contains("minecraft:pink")) {
+                    bedcolor = 6;
+                } else if (javaId.contains("minecraft:gray")) {
+                    bedcolor = 7;
+                } else if (javaId.contains("minecraft:light_gray")) {
+                    bedcolor = 8;
+                } else if (javaId.contains("minecraft:cyan")) {
+                    bedcolor = 9;
+                } else if (javaId.contains("minecraft:purple")) {
+                    bedcolor = 10;
+                } else if (javaId.contains("minecraft:blue")) {
+                    bedcolor = 11;
+                } else if (javaId.contains("minecraft:brown")) {
+                    bedcolor = 12;
+                } else if (javaId.contains("minecraft:green")) {
+                    bedcolor = 13;
+                } else if (javaId.contains("minecraft:red")) {
+                    bedcolor = 14;
+                } else if (javaId.contains("minecraft:black")) {
+                    bedcolor = 15;
+                }
```

Comment:
Instead of having this big if chain would it not be cleaner to store the bed color in `blocks.json`, this would be an easy addition to the mappings-generator

Reply:
Implemented with GeyserMC/mappings-generator#5. This code will get updated depending on the status of that pull request.